### DISCUSSION TITLE
Harmonization of highlights

### DISCUSSION THREAD

**Discussion by polo2137** [koreader/koreader#12096] [Ideas]
Hello,

First and foremost I love koreader and really enjoy using it so I became motivated to help with improving it for others.

Recently I've been reading a lot of books about the Polish law and the highlight feature is marvellous.
Although I really enjoy it there is inconsistency that I detected with how it works which drives me nuts.

When highlighting a piece of text **usually** koreader also automatically includes the last signs of a sentence if highlighting a word at the ending of the sentence, eg. _)._ , _._ , _"._ and so on. This is the very useful feature making highlighting fast and simple. There is a catch though - it doesn't work 100% of the time but more like 80%. 

Aforementioned behaviour doesn't occur at all while taking advantage of afterward highlighting using popup window with four arrows that appears as a result of a click on highlighted text. Moreover with above-mentioned functionality there is no possibility to highlight the remaining orphan characters because the highlight jumps to the first word in a new sentence.

The behaviour described above sometimes makes highlighting a frustrating experience when you have to delete a long highlight and then select whole important text again.

Device: Kobo Libra 2
Book format: epub

I would be extremely glad if you would find some time to fix described discrepancies.

**Comment by poire-z** [comment]
What you describe are the expected behaviours, nothing to really fix. And I think it's documented in the User Guide.

> it doesn't work 100% of the time but more like 80%.

See links and other comments in https://github.com/koreader/koreader/issues/8857#issuecomment-1053711400 (first issue I found, may not be the latest).

> Moreover with above-mentioned functionality there is no possibility to highlight the remaining orphan characters because the highlight jumps to the first word in a new sentence.

Long press on the arrow symbol to move by char - tap move by word indeed.
